pkg/kubectl/resource/mapper.go

```diff
@@ -50,8 +52,16 @@ type Mapper struct {
 func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
 	versions := &runtime.VersionedObjects{}
 	_, gvk, err := m.Decode(data, nil, versions)
+	var obj runtime.Object
+	var versioned runtime.Object
+	if registered.IsThirdPartyAPIGroupVersion(gvk.GroupVersion()) {
+		obj, err = runtime.Decode(thirdpartyresourcedata.NewCodec(nil, gvk.Kind), data)
+		versioned = obj
```

Comment:
`kubectl apply ...` doesn't appear to work.  I would like to get this PR in and then add an issue to fix that particular problem, unless you see a clean fix in this particular code.  (I don't)